test/enn_undersample.jl

```diff
@@ -0,0 +1,132 @@
+using Imbalance: compute_enn_filter
+
+@testset "filter computation" begin
+
+    # test computing the filter
+    X = [1 2; 1.1 2.1; 1.05 2.05; 1.01 2.01; 2 3; 2.1 3.1; 2.05 3.05; 2.01 3.01]
+    y = [1, 1, 1, 2, 2, 2, 2, 3]
+    @test compute_enn_filter(X', y, 3, "mode") == [1, 1, 1, 0, 1, 1, 1, 0]
+
+    X = [
+        1 2
+        1.1 2.1
+        1.05 2.05
+        1.01 2.01
+        1.011 2.011
+        2 3
+        2.1 3.1
+        2.05 3.05
+        2.01 3.01
+        2.011 3.011
+    ]
+    y = [1, 1, 1, 2, 2, 2, 2, 2, 3, 2]
+    @test compute_enn_filter(X', y, 4, "only mode") == [0, 0, 0, 0, 0, 1, 1, 1, 0, 1]
+    @test compute_enn_filter(X', y, 4, "exists") == [1, 1, 1, 1, 1, 1, 1, 1, 0, 1]
+
+    X = [
+        1 2
+        1.1 2.1
+        1.05 2.05
+        1.01 2.01
+        1.011 2.011
+        2 3
+        2.1 3.1
+        2.05 3.05
+        2.01 3.01
+        2.011 3.011
+    ]
+    y = [1, 1, 1, 1, 1, 2, 2, 2, 3, 2]
+    @test compute_enn_filter(X', y, 4, "all") == [1, 1, 1, 1, 1, 0, 0, 0, 0, 0]
+end
+
+@testset "end-to-end test with imbalanced-learn" begin
+    ENV["PYTHON"] = ""
+    import Pkg
+    Pkg.build("PyCall")
+    using PyCall
+    using Conda
+
```

Comment:
This is nice, but is it possible to do the scikit learn test off-line and just paste in the results here? Adding the PyCall adds some overhead, and any loading of python libraries is less robust than pure Julia testing or testing using pre-built binaries.

Reply:
Thanks for the recommendation. PyCall has been a true pain because its tests take centuries to finish.

Reply:
@ablaom okay, I tried giving this a shot and it doesn't seem that easy. For instance, in one test I loop over each one of eight columns and check that the column's distance matrix is correct. Each matrix is already big, it would be tedious to copy and paste them into the test.

> I can imagine maintenance headaches long term. For example, you are not locking in a particular version of the python libraries here, right?

Yes, thanks for nothing that. I will lock them so this is no longer a problem.

Reply:
Done.

Reply:
Okay. I guess we can simplify the tests if it becomes a hassle later.